src/prefect/engine/flow_runner.py

```diff
@@ -566,11 +566,8 @@ def run_task(
 
         """
         with prefect.context(self.context):
-            default_handler = task.result_handler or self.flow.result_handler
             task_runner = self.task_runner_cls(
-                task=task,
-                state_handlers=task_runner_state_handlers,
-                result_handler=default_handler,
```

Comment:
Yea I think we should re-introduce this init logic for results

Reply:
Reintroduced result_handler forwarding (for backwards compatability/conversion) in  af59d67 and added in result forwarding in e1f8e2f